homeassistant/components/esphome/sensor.py

```diff
@@ -104,18 +105,18 @@ def device_class(self) -> str | None:
         return self._static_info.device_class
 
     @property
-    def state_class(self) -> str | None:
+    def state_class(self) -> SensorStateClass | str | None:
```

Comment:
This won't return `str` anymore?

Reply:
You're right - I have adjusted the type hints.